accounts/usbwallet/wallet.go

```diff
@@ -524,6 +526,45 @@ func (w *wallet) signHash(account accounts.Account, hash []byte) ([]byte, error)
 
 // SignData signs keccak256(data). The mimetype parameter describes the type of data being signed
 func (w *wallet) SignData(account accounts.Account, mimeType string, data []byte) ([]byte, error) {
+
+	// dispatch to 712 signing if the mimetype is TypedData and the format matches
+	if mimeType == accounts.MimetypeTypedData && data[0] == 0x19 && data[1] == 0x01 && len(data) == 66 {
```

Comment:
Please do a length-check on `data` before directly accessing elements, otherwise it's a potential crash

Reply:
If I understand correctly, golang short circuits boolean expressions, so I addressed this by reordering the conditions